### PR TITLE
Remove options from the `test-title` rule

### LINKED ISSUE #185: Remove options from the `test-title` rule
[Issuehunt badges](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/185)

When https://github.com/avajs/ava/pull/1669 ships tests will always be required to have a title. This makes the [current `test-title` options](https://github.com/avajs/eslint-plugin-ava/blob/3176c84c38c23616cc79193fa4b97a40456af476/docs/rules/test-title.md#options) obsolete.

> There is a $30.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/185).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/avajs/eslint-plugin-ava/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

### PR DESCRIPTION
**Pull request by GMartigny**
fix #185

### CODE REVIEW

**Comment by GMartigny** [comment]
Hi,
First draft to be sure I'm going the right way. Tests have been updated, but not docs yet.
As per https://github.com/avajs/eslint-plugin-ava/issues/185#issuecomment-361262898, the rule only errors when there's no argument or the first one is a function.

**Comment by sindresorhus** [comment]
The implementation direction looks good to me.

File: docs/rules/test-title.md [hunk 1]
```diff
@@ -2,24 +2,12 @@
 
 Translations: [Français](https://github.com/avajs/ava-docs/blob/master/fr_FR/related/eslint-plugin-ava/docs/rules/test-title.md)
 
-Tests should have a title.
+Tests should have a title, because test without title would be rejected by AVA since [v1.0.1](https://github.com/avajs/ava/releases/tag/v1.0.1).
```
Comment (sindresorhus) [inline docs/rules/test-title.md]:
```suggestion
Tests should have a title. AVA [v1.0.1](https://github.com/avajs/ava/releases/tag/v1.0.1) and later enforces this at runtime.
```

**Comment by sindresorhus** [comment]
Is this ready?

**Comment by GMartigny** [comment]
> Is this ready?

It is for me.

**Pull request merged**